subsys/bluetooth/host/conn.c

```diff
@@ -2835,6 +2835,24 @@ int bt_conn_auth_cb_register(const struct bt_conn_auth_cb *cb)
 	return 0;
 }
 
+int bt_conn_auth_cb_overlay(struct bt_conn *conn, const struct bt_conn_auth_cb *cb)
+{
+	/* The cancel callback must always be provided if the app provides
+	 * interactive callbacks.
+	 */
+	if (!cb->cancel &&
```

Comment:
Suggest to use `CHECKKIF` for static checks like these that can always be verified by the caller before calling